src/QsCompiler/Compiler/Templates/QirDriverCpp.tt

```diff
@@ -1,16 +1,342 @@
 ﻿<#@ template language="C#" #>
+<#@ import namespace="Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint" #>
 //------------------------------------------------------------------------------
 // This code was generated by a tool.
 // <auto-generated />
 //------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
 
-// N.B. This is currently just a base non-functional example of a template used
-//      to generate C++ code that works as an entry-point driver for a QIR-based
-//      executable.
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <vector>
 
-extern "C" void <#= entryPointOperation.Name #>(); // NOLINT
+#include "CLI11.hpp"
+
+#include "QirContext.hpp"
+#include "QirRuntime.hpp"
+#include "SimFactory.hpp"
+
+using namespace Microsoft::Quantum;
+using namespace std;
+
+struct InteropArray
```

Comment:
Eventually we will move some of the generic logic and types to its own library and header, but in the meantime it would be great to not include types and logic that are not needed in the generated code.

For this case, extending the `EntryPointOperation` class to have a `ContainsArgumentType(DataType type)` method would be very useful.

Reply:
removed if not used now

Reply:
Can you please do the same for the other types defined here (e.g. `InteropRange` which is only used when a Q# `Range` type argument is present?

Reply:
done

Reply:
This can also be applied to auxiliary functions and data structures. More specifically:
- `InteropFalseAsChar`, `InteropTrueAsChar` and `BoolAsCharMap` are not needed if there are no Bool or Bool array arguments.
- `PauliMap` and `TranslatePauliToChar` are not needed if there are no Pauli or Pauli array arguments.
- `InteropResultZeroAsChar`, `InteropResultOneAsChar` and `ResultAsCharMap` are not needed if there are no Result or Result array arguments.
- `CreateInteropArray` and `TranslateVector` are not needed if there are no array arguments.
- `CreateInteropRange` and `TranslateRangeTupleToInteropRangePointer` are not needed if there are no Range or Range array arguments.
- `TranslateStringToCharBuffer` is not needed if there are no String or String array arguments.
- `FreePointerVector` is not needed if there are no Range array arguments.

Reply:
done